src/lib/concurrency/logging/binary_recovery.cpp

```diff
@@ -0,0 +1,156 @@
+#include "binary_recovery.hpp"
+
+#include <fstream>
+
+#include "../../operators/insert.hpp"
+#include "../../storage/storage_manager.hpp"
+#include "../../storage/table.hpp"
+#include "../transaction_manager.hpp"
+#include "logger.hpp"
+#include "types.hpp"
+
+namespace opossum {
+
+BinaryRecovery& BinaryRecovery::getInstance() {
+  static BinaryRecovery instance;
+  return instance;
+}
+
+template <>
+std::string BinaryRecovery::_read(std::ifstream& file) {
+  std::string result;
+  std::getline(file, result, '\0');
+  return result;
+}
+
+AllTypeVariant BinaryRecovery::_read_AllTypeVariant(std::ifstream& file, DataType data_type) {
+  AllTypeVariant value;
+  switch (data_type) {
+    case DataType::Int: {
+      return AllTypeVariant{_read<int32_t>(file)};
+      break;
+    }
+    case DataType::Long: {
+      return AllTypeVariant{_read<int64_t>(file)};
+      break;
+    }
+    case DataType::Float: {
+      return AllTypeVariant{_read<float>(file)};
+      break;
+    }
+    case DataType::Double: {
+      return AllTypeVariant{_read<double>(file)};
+      break;
+    }
+    case DataType::String: {
```

Comment:
We try to keep the number of places where all supported data types are listed to a minimum. Any way to make this more general?